backend/src/main/validation/PromotionValidation.ts

```diff
@@ -12,6 +14,16 @@ export class PromotionValidation {
     userId: IdValidation.schema,
     placeId: Joi.string().required(),
     discount: DiscountValidation.schema,
+    /**
+     * A promotion must have at least one schedule and can have at most seven (each representing one day of the week)
+     * Will validate each item in array with ScheduleValidation's schema
+     * */
+    schedules: Joi.array()
+      .unique('dayOfWeek', { ignoreUndefined: true }) // ignore undefined so that in case object does not contain dayOfWeek at all
+      .min(1)
+      .max(7)
+      .items(ScheduleValidation.schema)
+      .required(),
```

Comment:
Let me know if this is okay! To me, I did not see a point for a user having more than 7 schedules.

Also, the validation message is not user friendly when we are checking for duplicate dayOfweek's for the array of schedules. It says something liek `schedules[1] contains a duplicate value`. In the future I can imagine something like this:

```
fun validate(target: any) {
 const errors = ... // get all validation errors from Joi
 errors.addAll(PromotionDTO.validateMoreComplicatedStuff(target as PromotionDTO))
 return errors;
}
```

That way `validateMoreComplicatedStuff` can contain custom logic for more complicated validations and has more user-friendly error messages. But at this point, the purpose of our validation is just to make sure we don't use DB transactions to tell us whether anything is ill-formatted. Better user-friendly messages should be another issue.